x/blockdb/entry_cache_test.go

```diff
@@ -0,0 +1,107 @@
+// Copyright (C) 2019-2025, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package blockdb
+
+import (
+	"slices"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestCacheOnMiss(t *testing.T) {
+	db, _ := newTestDatabase(t, DefaultConfig())
+	height := uint64(20)
+	block := randomBlock(t)
+	require.NoError(t, db.Put(height, block))
+
+	// Evict the entry from cache to simulate a cache miss
+	db.entryCache.Evict(height)
+
+	// Read the block - should populate the cache on cache miss
+	_, err := db.Get(height)
+	require.NoError(t, err)
+
+	_, ok := db.entryCache.Get(height)
+	require.True(t, ok)
+}
+
+func TestCacheHas(t *testing.T) {
+	db, _ := newTestDatabase(t, DefaultConfig())
+	height := uint64(30)
+	block := randomBlock(t)
+	require.NoError(t, db.Put(height, block))
+
+	has, err := db.Has(height)
+	require.NoError(t, err)
+	require.True(t, has)
+
+	// Verify block is in cache
+	cached, ok := db.entryCache.Get(height)
```

Comment:
this test doesn't really test that we don't do I/O and use the cache. 

what you can do is populate the cache artificially with a block that is not in the ledger and then Get/Has that block.